### PR TITLE
Rename event serializers

### LINKED ISSUE #2293: Rename EventSerializers to AgentEventSerializers
# Refactor

## Component(s) to be refactored

- event_serializers
- EventSerializerRegistry
- IEventSerializer
- PydanticEventSerializer

## Explanation
The Island handles both Island events and Agent events. To avoid confusion, rename event serializer components to include "agent" in their names. Don't forget to also rename the unit test files.

## Tasks
- [x] Rename `IEventSerializer` to `IAgentEventSerializer` (0d)
- [x] Rename `EventSerializerRegistry` to `AgentEventSerializerRegistry` (0d)
- [x] Rename `PydanticEventSerializer` to `PydanticAgentEventSerializer` (0d)
- [ ] Rename `event_serializers` package to `agent_event_serializers` (0.25d)

### PR DESCRIPTION
**Pull request by shreyamalviya**
# What does this PR do?

Fixes #2293 

## PR Checklist
* [x] Have you added an explanation of what your changes do and why you'd like to include them?
* [x] Is the TravisCI build passing?
* [ ] ~Was the CHANGELOG.md updated to reflect the changes?~
* [ ] ~Was the documentation framework updated to reflect the changes?~
* [x] Have you checked that you haven't introduced any duplicate code?

## Testing Checklist

* [ ] ~Added relevant unit tests?~
* [x] Have you successfully tested your changes locally? Elaborate:
    > UTs pass
* [ ] ~If applicable, add screenshots or log transcripts of the feature working~

### CODE REVIEW

**Review by VakarisZ — COMMENTED** [review]
Maybe rename `monkey/common/events` to `monkey/common/agent_events`?

**Comment by VakarisZ** [comment]
`monkey/infection_monkey/agent_event_forwarder.py:87`
logger.debug(f"Sending **agent** events to Island at {self._server_address}: {events}")

**Comment by VakarisZ** [comment]
Maybe the resource should also be called `AgentEvents`, since it's the only thing it accepts? Alternativelly, maybe it can have `/api/events/<string:type>` url?

**Comment by VakarisZ** [comment]
`monkey/monkey_island/cc/repository/i_event_repository.py` should be called `i_agent_event_repository.py`?

**Comment by mssalvatore** [comment]
I agree with @VakarisZ . That being said, the issue was fairly tightly scoped. I'd say fix as many of those things as you can in 30 minutes and create a new issue for the rest.

**Review by mssalvatore — APPROVED** [review]

File: vulture_allowlist.py [hunk 1]
```diff
@@ -18,16 +18,16 @@
...
 from monkey_island.cc.repository.i_attack_repository import IAttackRepository
 from monkey_island.cc.repository.i_config_repository import IConfigRepository
-from monkey_island.cc.repository.i_event_repository import IEventRepository
 from monkey_island.cc.repository.i_log_repository import ILogRepository
 from monkey_island.cc.repository.i_machine_repository import IMachineRepository
 from monkey_island.cc.repository.i_report_repository import IReportRepository
 from monkey_island.cc.repository.i_simulation_repository import ISimulationRepository
 from monkey_island.cc.repository.ICredentials import ICredentialsRepository
-from monkey_island.cc.repository.zero_trust.IEventRepository import IEventRepository
+from monkey_island.cc.repository.zero_trust.IAgentEventRepository import IAgentEventRepository
```
Comment (VakarisZ) [inline vulture_allowlist.py:30]: *happy-noises*

**Review by VakarisZ — APPROVED** [review]

**Pull request merged**